openverse_catalog/dags/common/loader/reporting.py

```diff
@@ -0,0 +1,24 @@
+import logging
+
+from common.slack import send_message
+
+
+logger = logging.getLogger(__name__)
+
+
+def report_completion(provider_name, media_type, duration, record_count):
+    """
+    Send a Slack notification when the load_data task has completed.
+    Messages are only sent out in production and if a Slack connection is defined.
+    In all cases the data is logged.
+    """
+    message = f"""
+*Provider*: `{provider_name}`
+*Media Type*: `{media_type}`
+*Number of Records Upserted*: {record_count}
+*Duration of data pull task*: {duration}
```

Comment:
Are there other scenarios where this could be _None_ besides the task being marked as done manually? Only asking because if there _aren't_ perhaps we could use a more verbose default instead of "None", something like "Task was ended early" or something (better than that, it's late for me haha)

Reply:
Sorry, took me awhile to get my environment ready to test -- yeah, I think something like this would be good. I believe this should only happen when the task was ended manually, since it needs to have been interrupted but still in a `success` state for this task to even run.

I think `duration` won't be falsy, though, it'll be the actual string `None` returned by xcom.

Reply:
Oh shoot, seriously?? :sob: I could have sworn I've gotten Airflow to handle Python objects as XComs appropriately... maybe we need [`render_template_as_native_obj` set to `True`?](https://airflow.apache.org/docs/apache-airflow/stable/concepts/operators.html#rendering-fields-as-native-python-objects)

Reply:
OH NOW I REMEMBER - we need to turn on [XCom pickling!](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#enable-xcom-pickling) I'll make an issue for it :smile: